wgsl/index.bs

```diff
@@ -5480,6 +5479,74 @@ See [[#sync-builtin-functions]].
     <td>Bitwise-exclusive-or. [=Component-wise=] when |T| is a vector.
 </table>
 
+
+<table class='data'>
+  <caption>Bit shift expressions</caption>
+  <thead>
+    <tr><th>Precondition<th>Conclusion<th>Notes
+  </thead>
+
+  <tr algorithm="concrete shift left">
+    <td>|e1|: |T|<br>
+    |e2|: |TS|<br>
+    |S| is [INT]<br>
+    |T| is |S| or vec|N|&lt;|S|&gt;<br>
+    |TS| is u32 when |T| is |S|, otherwise |TS| is vec|N|&lt;u32&gt;
+    <td class="nowrap">|e1| `<<` |e2|: |T|
+    <td>Shift left (concrete):
+
+        Shift |e1| left, inserting zero bits at the least significant positions,
+        and discarding the most significant bits.
+
+        The number of bits to shift is the value of |e2|, modulo the bit width of |e1|.<br>
+
+        [=Component-wise=] when |T| is a vector.
+
+  <tr algorithm="abstract shift left">
+    <td>|e1|: |T|<br>
+    |e2|: |T|<br>
+    |T| is AbstractInt or vec|N|&lt;AbstractInt&gt;
+    <td class="nowrap">|e1| `<<` |e2|: |T|
+    <td>Shift left (abstract):
+
+        Shift |e1| left, inserting zero bits at the least significant positions,
+        and discarding the most significant bits.
+
+        The number of bits to shift is the value of |e2|.
+
+        [=Component-wise=] when |T| is a vector.
+
+        It is a [=shader-creation error=] if the calculation would overflow the AbstractInt type.
```

Comment:
We're trying to allow a future where AbstractInt has more than 64 bits.  That's why we don't allow overflow.

So we don't want to have a calculation depend on "the" bit width of AbstractInt.

Reply:
Are you implying that an abstract int, in the future, could have multiple bit widths? Otherwise there is aways `a` bit-width for an abstract int so saying `the bit-width of abstract int` should always be correct regardless if abstract int is int8_t, int64_t or int256_t, no?

Reply:
I see, and a shader which previous did `1 << 65` and ended up doing a `1<<1` will now actually `1 << 65`